File: src/main/java/src/SistemaGestionEstudiante/Curso.java
package SistemaGestionEstudiante;

public class Curso {
    private String nombre;
    private Profesor profesor;

    public Curso(String nombre, Profesor profesor) {
        this.nombre = nombre;
        this.profesor = profesor;
        profesor.asignarCurso(this);
    }

    public String getNombre() {
        return nombre;
    }

    public Profesor getProfesor() {
        return profesor;
    }

    @Override
    public String toString() {
        return "Curso: " + nombre + ", Profesor: " + profesor.getNombre();
    }
}

File: src/main/kotlin/SistemaGestionEstudiante/Curso.kt
package org.example.SistemaGestionEstudiante

class Curso(val nombre: String, profesor: Profesor) {
    private val profesor: Profesor = profesor

    init {
        profesor.asignarCurso(this)
    }

    override fun toString(): String {
        return "Curso: " + nombre + ", Profesor: " + profesor.nombre
    }
}